### PR TITLE
Deallocate arrays with proper destructor

### PR DESCRIPTION
**Pull request by tparys**
Signed-off-by: Timothy Parys <tparys@gmail.com>

### CODE REVIEW

**Comment by tparys** [comment]
Something got badly out of sequence while trying to fix #113, so trying again.

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/92/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux%20aarch64/46/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/38/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/92/)

File: src/cpp/FastCdr.cpp [hunk 1]
```diff
@@ -777,7 +777,7 @@ FastCdr& FastCdr::deserializeStringSequence(
     catch (eprosima::fastcdr::exception::Exception& ex)
     {
         free(sequence_t);
-        sequence_t = NULL;
+        delete [] sequence_t;
```
Comment (MiguelCompany) [inline src/cpp/FastCdr.cpp:780]: I think you wanted to change the previous line, not this one.

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

File: src/cpp/FastCdr.cpp (hunk 1 above)
Comment (tparys) [reply src/cpp/FastCdr.cpp:780]: Good catch. Will fix.

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/93/)
* Linux aarch64[Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux%20aarch64/47/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Mac/39/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/93/)

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Comment by MiguelCompany** [comment]
@tparys Thanks for your contribution!

**Pull request merged**